static/index.js

```diff
@@ -0,0 +1,136 @@
+  {
+    review: "Cool, cool, cool, cool. No doubt, no doubt, no doubt.",
+    rating: 5,
+  },
+];
+
+function saveToLocalStorage(reviews) {
+  const reviewsString = JSON.stringify(reviews);
+  localStorage.setItem("reviews-list", reviewsString);
+}
+
+function loadFromLocalStorage() {
+  const reviewsString = localStorage.getItem("reviews-list");
+
+  return JSON.parse(reviewsString);
+}
+
+function renderReviewsList() {
+  const reviewRatingListElement = document.getElementById(
+    "reviews-ratings-list"
+  );
+  reviewRatingListElement.innerHTML = "";
+  reviewsList.forEach((review) => {
+    reviewRatingListElement.appendChild(createReview(review));
+  });
+
+  calculateAverageRating();
+  saveToLocalStorage(reviewsList);
+}
+
+function createReview(review) {
+  const reviewRatingElement = document.createElement("div");
+  reviewRatingElement.id = "review-rating-item";
+
+  const reviewElement = document.createElement("span");
+  reviewElement.textContent = review.review;
+  reviewRatingElement.appendChild(reviewElement);
+
+  const ratingElement = document.createElement("span");
+  ratingElement.textContent = review.rating + " / 5";
+  reviewRatingElement.appendChild(ratingElement);
+
+  const ratingStarsElement = document.createElement("div");
+  ratingStarsElement.id = "rating-stars";
+  for (let i = 1; i <= 5; i++) {
+    const ratingStar = document.createElement("span");
+    ratingStar.textContent = "☆";
+    if (i <= review.rating) {
+      ratingStar.style.color = "gold";
+    }
+    ratingStarsElement.appendChild(ratingStar);
+  }
+  reviewRatingElement.appendChild(ratingStarsElement);
+
+  const deleteButton = document.createElement("button");
+  deleteButton.textContent = "Remove";
+  deleteButton.onclick = () => {
+    reviewsList = reviewsList.filter((r) => {
+      return r !== review;
+    });
+    renderReviewsList();
+  };
+  reviewRatingElement.appendChild(deleteButton);
+
+  return reviewRatingElement;
+}
+
+const newReviewButtonHandler = () => {
+  const reviewInput = document.getElementById("review");
+  const ratingInputList = document.querySelectorAll(".star");
+  let ratingValue = 0;
+  ratingInputList.forEach((ratingInput) => {
+    if (ratingInput.style.color === "gold") {
+      ratingInput.style.color = "gray";
+      ratingValue += 1;
+    }
+  });
+
+  if (!reviewInput.value || !ratingValue) {
+    return;
+  }
+
+  const newReview = {
+    review: reviewInput.value,
+    rating: ratingValue,
+  };
+
+  reviewsList.push(newReview);
+  renderReviewsList();
+  reviewInput.value = "";
+};
+
+function calculateAverageRating() {
+  let averageRating = 0;
+  reviewsList.forEach((review) => {
+    averageRating += Number(review.rating);
+  });
+  if (!reviewsList.length) {
+    averageRating = 0;
+  } else {
+    averageRating /= reviewsList.length;
+    averageRating = averageRating.toPrecision(2);
+  }
+
+  const averageRatingElement = document.getElementById("average-rating");
+  if (averageRating > 0) {
```

Comment:
Mrvicu overengineer za ovo.
Mogao si ovo rješti u nekoliko linija:
```
let averageRating = 0;

reviewsList.forEach((review) => {
    averageRating += parseInt(review.rating)
})

  const averageRatingElement = document.getElementById("average-rating");
  averageRatingElement.textContent = averageRating > 0 ? averageRating.toFixed(1) + " / 5": "There are no ratings currently.";
```
Ukoliko je reviewsList prazan array, forEach je dovoljno pametan da ne prolazi kroz petlju tako da će to odmah preskočiti :)
Prođemo na predaji još oko ovoga :)